parquet-column/src/main/java/org/apache/parquet/column/values/bytestreamsplit/ByteStreamSplitValuesReaderForDouble.java

```diff
@@ -0,0 +1,51 @@
+/* 
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.parquet.column.values.bytestreamsplit;
+
+import org.apache.parquet.io.ParquetDecodingException;
+
+import java.io.IOException;
+
+public class ByteStreamSplitValuesReaderForDouble extends ByteStreamSplitValuesReader {
+
+  private byte[] valueByteBuffer;
+
+  public ByteStreamSplitValuesReaderForDouble() {
+    super(Double.BYTES);
+    valueByteBuffer = new byte[Double.BYTES];
+  }
+
+  @Override
+  public double readDouble() {
+    try {
+      gatherElementDataFromStreams(valueByteBuffer);
+      long value = (long)(valueByteBuffer[0] & 0xFF) |
```

Comment:
I think ByteBuffer.wrap(bytes).order().getDouble() should can do the work. You don't need to reimplement it.

Reply:
Changed it but I didn't modify the tests, which test whether the Encoding or Decoding paths produce correct results, to rely on the ByteBuffer serialization/deserialization.